### PR TITLE
WIP: test_gap_in_public_chat

### PR DESCRIPTION
**Pull request by pavloburykh**
This PR adds autotest for testcase https://ethstatus.testrail.net/index.php?/cases/view/700727

### CODE REVIEW

File: test/appium/tests/atomic/chats/test_public.py [hunk 1]
```diff
@@ -435,6 +435,38 @@ def test_different_status_in_timeline(self):
...
+        device_1, device_2 = SignInView(self.drivers[0]), SignInView(self.drivers[1])
+        home_1, home_2 = device_1.create_user(), device_2.create_user()
+        profile_1, profile_2 = home_1.profile_button.click(), home_2.profile_button.click()
+        profile_1.sync_settings_button.click()
+        profile_2.sync_settings_button.click()
+        profile_1.sync_history_for_button.click()
+        profile_2.sync_history_for_button.click()
+        profile_1.element_by_translation_id("two-minutes").click()
+        profile_2.element_by_translation_id("two-minutes").click()
+        profile_1.home_button.click()
```
Comment (churik) [inline test/appium/tests/atomic/chats/test_public.py]: loop please from 445-452

File: test/appium/tests/atomic/chats/test_public.py [hunk 2]
```diff
@@ -435,6 +435,38 @@ def test_different_status_in_timeline(self):
...
+        profile_2.home_button.click()
+        chat_1 = home_1.join_public_chat("pavlotest")
+        chat_2 = home_2.join_public_chat("pavlotest")
+        chat_1.send_message("HI")
+        # chat_1.home_button.click()
+        device_1.toggle_airplane_mode()
+        chat_2.send_message("testing gap")
```
Comment (churik) [inline test/appium/tests/atomic/chats/test_public.py]: "testing gap" should be variable

File: test/appium/tests/atomic/chats/test_public.py [hunk 3]
```diff
@@ -435,6 +435,38 @@ def test_different_status_in_timeline(self):
...
+        sleep(180)
+        device_1.toggle_airplane_mode()
+        sleep(120)
```
Comment (churik) [inline test/appium/tests/atomic/chats/test_public.py]: what is the reason for waiting 120 sec more?

File: test/appium/tests/atomic/chats/test_public.py [hunk 4]
```diff
@@ -435,6 +435,38 @@ def test_different_status_in_timeline(self):
...
+        chat_1.element_by_translation_id("fetch-messages").click()
+        chat_1.chat_element_by_text("testing gap")
```
Comment (churik) [inline test/appium/tests/atomic/chats/test_public.py]: There is no actual check.
Should be is_element_displayed in particular case

**Review by churik — CHANGES_REQUESTED** [review]

**Comment by pavloburykh** [comment]
@churik Thanx a lot for review and comments. I've pushed fixes, please take a look.

File: test/appium/tests/atomic/chats/test_public.py (hunk 1 above)
Comment (pavloburykh) [reply test/appium/tests/atomic/chats/test_public.py]: thanx, @churik , fixed

File: test/appium/tests/atomic/chats/test_public.py (hunk 2 above)
Comment (pavloburykh) [reply test/appium/tests/atomic/chats/test_public.py]: thanx, @churik , fixed

File: test/appium/tests/atomic/chats/test_public.py (hunk 3 above)
Comment (pavloburykh) [reply test/appium/tests/atomic/chats/test_public.py]: thanx, @churik , replaced by wait_for_element_starts_with_text() method

File: test/appium/tests/atomic/chats/test_public.py (hunk 4 above)
Comment (pavloburykh) [reply test/appium/tests/atomic/chats/test_public.py]: thanx, @churik , added is_element_displayed() method for check

File: test/appium/tests/atomic/chats/test_public.py [hunk 5]
```diff
@@ -435,6 +435,35 @@ def test_different_status_in_timeline(self):
...
+            element.click()
+        home_1.home_button.click()
+        chat_1, chat_2 = home_1.join_public_chat(chat_name), home_2.join_public_chat(chat_name)
+        chat_1.send_message("HI")
+        device_1.toggle_airplane_mode()
+        chat_2.send_message(message)
+        sleep(180)
+        device_1.toggle_airplane_mode()
+        chat_1.wait_for_element_starts_with_text("↓ Fetch messages", 60)
+        chat_1.element_by_translation_id("fetch-messages").click()
```
Comment (churik) [inline test/appium/tests/atomic/chats/test_public.py]: replace with wait and click

File: test/appium/tests/atomic/chats/test_public.py [hunk 6]
```diff
@@ -435,6 +435,35 @@ def test_different_status_in_timeline(self):
...
+    @marks.testrail_id(700727)
+    @marks.medium
+    def test_gap_in_public_chat(self):
+        self.create_drivers(2)
+        device_1, device_2 = SignInView(self.drivers[0]), SignInView(self.drivers[1])
+        home_1, home_2 = device_1.create_user(), device_2.create_user()
+        message = "testing gap"
+        chat_name = home_1.get_random_chat_name()
+        profile_1 = home_1.profile_button.click()
+        for element in [profile_1.sync_settings_button, profile_1.sync_history_for_button, profile_1.element_by_translation_id("two-minutes")]:
```
Comment (churik) [inline test/appium/tests/atomic/chats/test_public.py]: better don't have a loop here for best visibility

**Review by churik — CHANGES_REQUESTED** [review]

File: test/appium/tests/atomic/chats/test_public.py (hunk 5 above)
Comment (pavloburykh) [reply test/appium/tests/atomic/chats/test_public.py]: thanx @churik , fixed

File: test/appium/tests/atomic/chats/test_public.py (hunk 6 above)
Comment (pavloburykh) [reply test/appium/tests/atomic/chats/test_public.py]: thanx @churik , fixed

**Comment by pavloburykh** [comment]
@churik thanx for the review and comment. Please, review my recent fixes that I pushed.

**Review by churik — APPROVED** [review]
🎉

**Pull request merged**